File: Algorithm/leetcode/41.缺失的第一个正数.go
/*
 * @lc app=leetcode.cn id=41 lang=golang
 *
 * [41] 缺失的第一个正数
 */
package main
import (
	"fmt"
)



// @lc code=start
func firstMissingPositive(nums [6]int) int {
	n := len(nums);
	for i:=0; i<n; i++{
		fmt.Println(nums[i])
	}
}
func main(){
	firstMissingPositive([1,2,3,4,56])
}

// @lc code=end

File: Algorithm/leetcode/41.缺失的第一个正数.js
/*
 * @lc app=leetcode.cn id=41 lang=javascript
 *
 * [41] 缺失的第一个正数
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {number}
 */
var firstMissingPositive = function (nums) {
  if (nums.length === 0) return 1;
  if (nums.length === 1) {
    return nums[0] === 1 ? 2 : 1;
  }
  const numsSet = new Set(nums);
  if (!numsSet.has(1)) return 1;
  for (let i = 1;i <= nums.length;i++) {
    if (!numsSet.has(i)) return i;
  }
  if (numsSet.has(nums.length)) {
    return nums.length + 1;
  }
  return nums.length
};
// @lc code=end

